server/src/main/java/org/elasticsearch/index/mapper/AbstractGeometryFieldMapper.java

```diff
@@ -143,7 +153,31 @@ public Builder ignoreZValue(final boolean ignoreZValue) {
 
     @Override
     protected Object parseSourceValue(Object value, String format) {
-        throw new UnsupportedOperationException();
+        AbstractGeometryFieldType<Parsed, Processed> mappedFieldType = fieldType();
+        Parser<Parsed> geometryParser = mappedFieldType.geometryParser();
+        Formatter<Parsed> geometryFormatter = mappedFieldType.geometryFormatter();
+
+        Parsed geometry;
+        try (XContentParser parser = new MapXContentParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE,
```

Comment:
This seemed like the easiest way to integrate with the existing logic but is far from ideal:
* We perform parsing even when it's not necessary: even when the geometry is already in the right format, we still parse it to an object re-serialize it.
* We also awkwardly translate between maps and xContent.

Let me know if you have suggestions around a better approach. Overall I was hoping to keep this PR reasonably scoped, since it is part of a large 'field retrieval' change that already has a few moving parts. But we could plan a larger refactor in a separate/ follow-up change.

Reply:
This is pretty yikes, but I do like the idea of keeping the work contained on the branch and doing a larger refactoring after merging. Maybe it'd be cleaner to push the ability to parse these fields into `SourceLookup`. That way we don't need to do the `xcontent -> map -> xcontent` dance.

Reply:
would it make sense to specialize between points and shapes? At least for Shapes, we should be able to tell whether `value` is an object or a string, right?

Reply:
> Maybe it'd be cleaner to push the ability to parse these fields into SourceLookup

I think this would be challenging given the current design of `SourceLookup`, since it is operates only on the level of xContent and is completely agnostic to the mappings. It would be nice though if `SourceLookup` could provide direct access to the xContent representation instead of always parsing + returning generic objects. This feels like a bigger change that's better suited for a follow-up though?

> At least for Shapes, we should be able to tell whether value is an object or a string, right?

This would work, I'll look into it further! I'm inclined to not worry about it though if it makes the design more complicated (specifically splitting points vs. shapes handling).